src/clp_ffi_js/ir/LogEventWithFilterData.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+#define CLP_FFI_JS_IR_LOGEVENTWITHFILTERDATA_HPP
+
+#include <utility>
+
+#include <clp/ir/types.hpp>
+
+#include <clp_ffi_js/constants.hpp>
+
+namespace clp_ffi_js::ir {
+/**
+ * A class which accepts a log event type as a template parameter and provides additional members
+ * for the log level and timestamp. The additional members facilitate log level filtering.
```

Comment:
I understand we are including the timestamp in this class because we want to enable timestamp parsing for the `StructuredStreamReader`; however, the intention of such is not very explicit in this PR.
Shall we say
```
A templated class that accepts a log event type and extends it with additional members for log level and timestamp. These members enable efficient filtering by log level and timestamp, enhancing flexibility in log processing.
```

Reply:
All good except I think you meant to capitalize `StreamReader`.

Reply:
Sorry I just read the latest code. I think that's good enough. (Kirk might have better ideas about improving that though